addons/pkg/util/condition_util.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	runtanzuv1alpha3 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha3"
+)
+
+const UnknownCondition = v1alpha1.AppConditionType("UnknownCondition")
+
+type state struct {
+	stateExists bool
+	status      corev1.ConditionStatus
+}
+
+func (s *state) setState(cond v1alpha1.AppCondition) {
+	s.stateExists = true
+	if s.status == "" || cond.Status == corev1.ConditionTrue {
+		s.status = cond.Status
+	} else if s.status == corev1.ConditionUnknown && cond.Status == corev1.ConditionFalse {
+		s.status = cond.Status
+	}
```

Comment:
question:
- why don't we always `s.status = cond.Status` ?
- how about other cases not caught by `if` and `else if` in above ?

Reply:
Cause there are 3 possible states: True, False, Unknown (any other state is not known and don't need to show; Also, with this if/else states overriding each other is honoured)

Reply:
Changed logic